collector/exporter/otelarrowexporter/internal/arrow/stream.go

```diff
@@ -396,6 +393,12 @@ func (s *Stream) processBatchStatus(status *arrowpb.BatchStatus) error {
 	ch, ret := s.getSenderChannels(status)
 
 	if ch == nil {
+		// This indicates the server received EOF from client shutdown.
+		// This is not an error because this is an expected shutdown
+		// initiated by the client by setting max_stream_lifetime.
+		if status.BatchId == -1 && status.StatusCode == arrowpb.StatusCode_OK {
```

Comment:
I suggest the use of a new status code to indicate this condition. While OK is fine, I think we shouldn't have to check `BatchId == -1` when this is happening, meaning we should check the status before calling `getSenderChannels()`. getSenderChannels is never going to return non-nil when BatchId is -1, since it's not a real batch there can't be any channel. So I'd check for `status.StatusCode == arrowpb.StatusCode_ShutdownPlease` or something like that above the call to `getSenderChannels()`.

Reply:
On reading the above, possibly we don't need to call `processBatchStatus()` at all when the streaming is shutting down, meaning to check around line 360 before calling `processBatchStatus()`.

Reply:
Thanks for the suggestions, this simplified things a lot. Let me know if you have any further suggestions